server/src/test/java/org/elasticsearch/ElasticsearchExceptionTests.java

```diff
@@ -610,6 +610,85 @@ public void testToXContent() throws IOException {
         }
     }
 
+    public void testGenerateFailureToXContentWithNoDetails() throws IOException {
+        {
+            Exception ex;
+            if (randomBoolean()) {
+                // just a wrapper which is omitted
+                ex = new RemoteTransportException("foobar", new FileNotFoundException("foo not found"));
+            } else {
+                ex = new FileNotFoundException("foo not found");
+            }
+            assertFailureAsJson(ex, """
+                {"error":{"type":"file_not_found_exception","reason":"foo not found"}}""", false);
+        }
+        {
+            ParsingException ex = new ParsingException(1, 2, "foobar", null);
+            assertFailureAsJson(ex, """
+                {"error":{"type":"parsing_exception","reason":"foobar"}}""", false);
+        }
+
+        { // render header and metadata
+            ParsingException ex = new ParsingException(1, 2, "foobar", null);
+            ex.addMetadata("es.test1", "value1");
```

Comment:
What's the significance of adding the metadata and the headers, should they be verified in the expected output if we get an exception? The comment says we render them, I'm probably not seeing how they are validated.

Reply:
Comment was left over from where I copied the exception from - updated